lib/middleware/request-handler.ts

```diff
@@ -138,3 +130,371 @@ export function onRequest(req: Interface.IOurRequest,
+        }
+
+        req.body.forEach((s: {'correlationId': number;
+                              'security': string;
+                              'fields': string[];
+                              'options'?: any }): void => {
+            // Check if all requests are valid
+            // The Subscribe request will proceed only if all subscriptions are valid
+            if (!_.has(s, 'correlationId') ||
+                !_.isNumber(s.correlationId) ||
+                !_.has(s, 'security') ||
+                !_.isString(s.security) ||
+                !_.has(s, 'fields') ||
+                !_.isArray(s.fields)) {
+                req.log.debug('Invalid subscription option.');
+                throw new Error('Invalid subscription option.');
+            }
+            if (req.apiSession.receivedSubscriptions.has(s.correlationId)) {
+                req.log.debug('Correlation Id ' + s.correlationId + ' already exist.');
+                throw new Error('Correlation Id ' + s.correlationId + ' already exist.');
+            }
+
+            var sub = new Subscription(s.correlationId,
+                                       s.security,
+                                       s.fields,
+                                       s.options,
+                                       conf.get('longpoll.maxbuffersize'));
+            subscriptions.push(sub);
+            req.apiSession.receivedSubscriptions.set(sub.correlationId, sub);
+
+            // Add event listener for each subscription
+            sub.on('data', (data: any): void => {
+                req.log.debug({data: {cid: sub.correlationId, time: process.hrtime()}},
+                             'Data received');
+
+                // Buffer the current data
+                sub.buffer.pushValue(data);
+            });
+        });
+
+        // Subscribe user request through blpapi-wrapper
+        return req.blpSession.subscribe(subscriptions);
+    })()
+        .then((): Promise<void> => {
+            if (!req.apiSession.expired) {
+                subscriptions.forEach((s: Subscription): void => {
+                    req.apiSession.activeSubscriptions.set(s.correlationId, s);
+                });
+                req.log.debug('Subscribed');
+                return res.sendEnd(0, 'Subscribed');
+            } else { // Unsubscribe if session already expires
+                try {
+                    req.blpSession.unsubscribe(subscriptions);
+                } catch (ex) {
+                    req.log.error(ex, 'Error Unsubscribing');
+                }
+                subscriptions.forEach((s: Subscription): void => {
+                    s.removeAllListeners();
+                    req.apiSession.receivedSubscriptions.delete(s.correlationId);
+                });
+                req.log.debug('Unsubscribed all active subscriptions');
+                return Promise.resolve();
+            }
+        })
+        .then(next)
+        .catch((err: Error): Promise<any> => {
```

Comment:
I guess it really should be `<any>` since this is the type definition for `next`:
```typescript
interface Next {
    (err?: any): any;
  }
```